### ISSUE TITLE
Magento\Customer\Model\Account\Redirect::setRedirectCookie() not properly working

### ISSUE THREAD

**Issue by nikoskip** [magento/magento2#24547]
### Preconditions (*)

1. Magento 2.3.2

### Steps to reproduce (*)

Call method ``Magento\Customer\Model\Account\Redirect::setRedirectCookie()``

    $this->accountRedirect->setRedirectCookie($this->urlBuilder->getUrl('checkout'));

### Expected result (*)

Cookie ``login_redirect`` created

### Actual result (*)

The cookie is sent to the browser, but not properly. All the cookie params are missing.

![image](https://user-images.githubusercontent.com/1230033/64664758-83683080-d426-11e9-9399-398681de70d1.png)

### Additional information

I checked the code and the problem is that the method ``setPublicCookie`` needs to include a ``PublicCookieMetadata`` instance as third param.

```
<?php
// vendor/magento/module-customer/Model/Account/Redirect.php:280
/**
     * Save redirect route to cookie for case of successful login/registration
     *
     * @param string $route
     * @return void
     */
    public function setRedirectCookie($route)
    {
        $this->getCookieManager()->setPublicCookie(self::LOGIN_REDIRECT_URL, $route);
    }
```
Indeed maybe ``setPublicCookie`` should require ``$metadata`` as mandatory. Why someone would need to create a cookie without params?

**Comment by ralphgarrison** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-20107` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by nikoskip** [comment]
BTW, the ``clearRedirectCookie`` is not working either. You also need to configure the cookie params to properly delete it.

I had to override this methods to get this to working. Here is my code if someone needs it, hope it helps!

```
<?php
    /**
     * @param string $route
     * @return void
     */
    public function setRedirectCookie($route)
    {
        $metadata = $this->cookieMetadataFactory->createPublicCookieMetadata();
        $metadata->setPath($this->sessionConfig->getCookiePath());
        $metadata->setDomain($this->sessionConfig->getCookieDomain());
        $metadata->setSecure($this->sessionConfig->getCookieSecure());
        $metadata->setHttpOnly($this->sessionConfig->getCookieHttpOnly());
        $metadata->setDuration($this->sessionConfig->getCookieLifetime());

        $this->getCookieManager()->setPublicCookie(self::LOGIN_REDIRECT_URL, $route, $metadata);
    }

    /**
     * @throws \Magento\Framework\Exception\InputException
     * @throws \Magento\Framework\Stdlib\Cookie\FailureToSendException
     */
    public function clearRedirectCookie()
    {
        $metadata = $this->cookieMetadataFactory->createPublicCookieMetadata();
        $metadata->setPath($this->sessionConfig->getCookiePath());
        $metadata->setDomain($this->sessionConfig->getCookieDomain());

        $this->getCookieManager()->deleteCookie(self::LOGIN_REDIRECT_URL, $metadata);
    }
```

**Comment by magento-engcom-team** [comment]
Hi @nikoskip. Thank you for your report. 
The issue has been fixed in magento/magento2#24612 by @sashas777 in 2.4-develop branch
Related commit(s):
 - [a15f33557d72d4ab05a99e9583fbfc0e2daa2ba2](https://github.com/magento/magento2/commit/a15f33557d72d4ab05a99e9583fbfc0e2daa2ba2)
 - [664056fc0e4fedc5c10aac2d14537b285dbefffd](https://github.com/magento/magento2/commit/664056fc0e4fedc5c10aac2d14537b285dbefffd)
 - [5188efdbcfe31e02147d2fc6846ffd5d40e4d0ca](https://github.com/magento/magento2/commit/5188efdbcfe31e02147d2fc6846ffd5d40e4d0ca)
 - [3cf8f224a3f850aa0838b700ba390eb25fd1f704](https://github.com/magento/magento2/commit/3cf8f224a3f850aa0838b700ba390eb25fd1f704)
 - [75970034babbf4525a2ccf5bf4ecfbe4d55fa897](https://github.com/magento/magento2/commit/75970034babbf4525a2ccf5bf4ecfbe4d55fa897)
 - [94138920f8c54365bc29f72b63b3a2de9c04a0a0](https://github.com/magento/magento2/commit/94138920f8c54365bc29f72b63b3a2de9c04a0a0)
 - [9f2bbc2c0060ce884c186a627403c1296080fe2d](https://github.com/magento/magento2/commit/9f2bbc2c0060ce884c186a627403c1296080fe2d)
 - [bfa636f75b220e3a59cb168ccc79b6ccdb055e07](https://github.com/magento/magento2/commit/bfa636f75b220e3a59cb168ccc79b6ccdb055e07)
 - [6446aed759f9ed44c1fecde4632e14f786b3117a](https://github.com/magento/magento2/commit/6446aed759f9ed44c1fecde4632e14f786b3117a)
 - [162ca604dae34980e972934bd4786756de52b0a1](https://github.com/magento/magento2/commit/162ca604dae34980e972934bd4786756de52b0a1)
 - [16da7068cdcc6077ff9ea2f58a0b20f56932e310](https://github.com/magento/magento2/commit/16da7068cdcc6077ff9ea2f58a0b20f56932e310)
 - [2c24ef9a6c15d342bc7109e09d1b5169482b3917](https://github.com/magento/magento2/commit/2c24ef9a6c15d342bc7109e09d1b5169482b3917)
 - [a514803313c20892020313c57fc50d336948d918](https://github.com/magento/magento2/commit/a514803313c20892020313c57fc50d336948d918)
 - [bfeadd7e3971a278784155cfddefc80c7aeeb07d](https://github.com/magento/magento2/commit/bfeadd7e3971a278784155cfddefc80c7aeeb07d)
 - [cd5ef3c32cfbcab0b080ab22e5c4a46f8f6b759c](https://github.com/magento/magento2/commit/cd5ef3c32cfbcab0b080ab22e5c4a46f8f6b759c)
 - [dc416a9bb545517384bff75c1bc1e181a1343af0](https://github.com/magento/magento2/commit/dc416a9bb545517384bff75c1bc1e181a1343af0)
 - [41ce59e5f53564e9e665660ccee618a390ca1de3](https://github.com/magento/magento2/commit/41ce59e5f53564e9e665660ccee618a390ca1de3)
 - [f021f1ef0514a9588eb439f689dcde99cef7525d](https://github.com/magento/magento2/commit/f021f1ef0514a9588eb439f689dcde99cef7525d)
 - [511e16a3ce006cb31621cb9b87cd576f0304c62b](https://github.com/magento/magento2/commit/511e16a3ce006cb31621cb9b87cd576f0304c62b)

The fix will be available with the upcoming 2.4.0 release.
